src/bin/pg_autoctl/cli_config.c

```diff
@@ -632,6 +632,157 @@ cli_config_set(int argc, char **argv)
 }
 
 
+static bool
+cli_keeper_config_validate_replication(KeeperConfig *config, PGSQL *pgsql)
+{
+	Keeper keeper = { 0 };
+	NodeAddress primary = { 0 };
+	bool primaryUnreachable;
+
+	if (!keeper_init(&keeper, config))
+	{
+		/* already logged */
+		return false;
+	}
+
+	/* get the primary info to populate the replication source */
+	if (!monitor_get_primary(&keeper.monitor,
+							 config->formation,
+							 keeper.state.current_group,
+							 &primary))
+	{
+		log_warn("Monitor could not be contacted for primary conninfo, skipping "
+				 "password validation");
+		return true;
+	}
+
+	if (primary.nodeId == keeper.state.current_node_id)
+	{
+		/* we are the primary */
+		PGSQL *pgsql = &keeper.postgres.sqlClient;
+
+		/* Skip validation if the primary isn't running. */
+		if (!pg_is_running(config->pgSetup.pg_ctl, config->pgSetup.pgdata))
+		{
+			log_warn("Primary could not be contacted, skipping password "
+					 "validation");
```

Comment:
Are we skipping the password validation, or the while `ALTER ROLE` operation altogether, and skipping the only actual change that we wanted to implement?

Reply:
w/ @jchampio
If the `ALTER ROLE` isn't issued (which means it was not possible to perform validation on the primary), we still write the config value to the file..so that the password change can be applied to the primary on config reload / restart in `keeper_ensure_configuration()`.